From 3c8e0802cb0e1db53bcb9a74ff31290c33d04b79 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Tue, 21 Feb 2023 07:08:46 +0100
Subject: [PATCH 24/32] ENH: Enable installation of Libs dev files

This enables the installation of development files (ui_.h, .h and .txx
files) for Slicer libraries MRML/Widgets, vtkITK and vtkSegmentationCore.
---
 Libs/ITKFactoryRegistration/CMakeLists.txt |  5 ++--
 Libs/MRML/Widgets/CMakeLists.txt           | 30 +++++++++++++++++++++-
 Libs/vtkITK/CMakeLists.txt                 |  5 ++--
 Libs/vtkSegmentationCore/CMakeLists.txt    |  5 ++--
 4 files changed, 38 insertions(+), 7 deletions(-)

diff --git a/Libs/ITKFactoryRegistration/CMakeLists.txt b/Libs/ITKFactoryRegistration/CMakeLists.txt
index 6f5c27bea9..ad290bb494 100644
--- a/Libs/ITKFactoryRegistration/CMakeLists.txt
+++ b/Libs/ITKFactoryRegistration/CMakeLists.txt
@@ -72,9 +72,10 @@ if(NOT DEFINED ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   set(${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT ON)
 endif()
 if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
-  file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.(h|txx)")
+  file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
+  file(GLOB templates "${CMAKE_CURRENT_SOURCE_DIR}/*.txx")
   install(
-    FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
+    FILES ${headers} ${templates} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
     DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
diff --git a/Libs/MRML/Widgets/CMakeLists.txt b/Libs/MRML/Widgets/CMakeLists.txt
index 25eabf41b4..c3c63c4091 100644
--- a/Libs/MRML/Widgets/CMakeLists.txt
+++ b/Libs/MRML/Widgets/CMakeLists.txt
@@ -434,8 +434,36 @@ if(NOT DEFINED ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
 endif()
 if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
+  # ui_ headers cannot be globbed at config time, here we list them explicitly
+  set(ui_headers
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLWindowLevelWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLVolumeThresholdWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLVolumeInfoWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLTransformSliders.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLThreeDViewInformationWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLThreeDViewControllerWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLTableViewControllerWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLSliceWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLSliceVerticalControllerWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLSliceInformationWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLSliceControllerWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLSegmentSelectorWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLScreenShotDialog.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLSceneFactoryWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLScalarsDisplayWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLROIWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLPlotViewControllerWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLNodeAttributeTableWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLNodeAttributeTableView.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLModelInfoWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLEventLoggerWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLDisplayNodeWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLColorPickerWidget.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/ui_qMRMLClipNodeWidget.h"
+  )
+
   install(
-    FILES ${headers}
+    FILES ${headers} ${ui_headers}
     DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
diff --git a/Libs/vtkITK/CMakeLists.txt b/Libs/vtkITK/CMakeLists.txt
index 3e0387ed64..6f74979e14 100644
--- a/Libs/vtkITK/CMakeLists.txt
+++ b/Libs/vtkITK/CMakeLists.txt
@@ -96,9 +96,10 @@ if(NOT DEFINED ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   set(${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT ON)
 endif()
 if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
-  file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.(h|txx)")
+  file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
+  file(GLOB templates "${CMAKE_CURRENT_SOURCE_DIR}/*.txx")
   install(
-    FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
+    FILES ${headers} ${templates} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
     DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
diff --git a/Libs/vtkSegmentationCore/CMakeLists.txt b/Libs/vtkSegmentationCore/CMakeLists.txt
index 4873578b55..97487ef003 100644
--- a/Libs/vtkSegmentationCore/CMakeLists.txt
+++ b/Libs/vtkSegmentationCore/CMakeLists.txt
@@ -20,9 +20,10 @@ if(NOT DEFINED ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   set(${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT ON)
 endif()
 if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
-  file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.(h|txx)")
+  file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
+  file(GLOB templates "${CMAKE_CURRENT_SOURCE_DIR}/*.txx")
   install(
-    FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
+    FILES ${headers} ${templates} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
     DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
-- 
2.39.3

